wire-ios-request-strategy/Sources/E2EIdentity/E2eIKeyPackageRotator.swift

```diff
@@ -0,0 +1,162 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireCoreCrypto
+import WireDataModel
+
+// sourcery: AutoMockable
+public protocol E2eIKeyPackageRotating {
+
+    func rotateKeysAndMigrateConversations(
+        e2eIService: E2eIServiceInterface,
```

Comment:
I'm wondering if we really need to inject `e2eIService`. We only use the `e2eIdentity` object

Reply:
We don't really need this except for tests